app/controllers/Controller.scala

```diff
@@ -104,7 +104,16 @@ class Controller @Inject() (
             allRepaymentData = vrtService.getAllRepaymentData(repaymentDetails, vrnMtd, financialData)
             engmtType = showResultsFormatter.computeEngmt(allRepaymentData)
             _ <- auditor.auditEngagement("showVrt", engmtType, Some(request.typedVrn.vrn))
-          } yield showResultsFormatter.computeView(allRepaymentData, customerData, vrnMtd)
+          } yield {
+            val inFlight = desFormatter.bankDetailsInFlight(customerData)
+            if (inFlight) {
+              val dateToDisplay = getDateToDisplay(customerData)
+              val welshDateToDisplay: String = dateToDisplay.welshMonth
+              showResultsFormatter.computeView(allRepaymentData, customerData, vrnMtd, Some(dateToDisplay), Some(welshDateToDisplay))
+            } else {
+              showResultsFormatter.computeView(allRepaymentData, customerData, vrnMtd, None, None)
```

Comment:
This could be technically simplified into the following:

```
val inFlight = desFormatter.bankDetailsInFlight(customerData)
val optDateToDisplay = if (inFlight) Some(getDateToDisplay(customerData)) else None

showResultsFormatter.computeView(allRepaymentData,customerData, vrnMtd,optDateToDisplay)
```
Not a huge blocker, but since the **welshMonth** is an implicit function for all Strings, it could just be called within the **repayment_details** partial.